EnviroSense.Application/Services/ApiKeyService.cs

```diff
@@ -0,0 +1,72 @@
+﻿using EnviroSense.Application.Algorithms;
+using EnviroSense.Application.Authorization;
+using EnviroSense.Domain.Entities;
+using EnviroSense.Repositories.Repositories;
+using Microsoft.Extensions.Logging;
+
+namespace EnviroSense.Application.Services;
+
+public class ApiKeyService : IApiKeyService
+{
+    private readonly IApiKeyRepository _apiKeyRepository;
+    private readonly IApiKeyGenerator _apiKeyGenerator;
+    private readonly IAuthorizationResolver _authorizationResolver;
+    private readonly ILogger<ApiKeyService> _logger;
+    private readonly IAccountService _accountService;
+
+    public ApiKeyService(
+        IApiKeyRepository apiKeyRepository,
+        IApiKeyGenerator apiKeyGenerator,
+        IAuthorizationResolver authorizationResolver,
+        ILogger<ApiKeyService> logger,
+        IAccountService accountService
+    )
+    {
+        _apiKeyRepository = apiKeyRepository;
+        _apiKeyGenerator = apiKeyGenerator;
+        _authorizationResolver = authorizationResolver;
+        _logger = logger;
+        _accountService = accountService;
+
+    }
+
+    public async Task<List<ApiKey>> List(Guid accountId)
+    {
+        return await _apiKeyRepository.ListAsync(accountId);
+    }
+
+    public async Task<ApiKey> GetByIdAsync(Guid Id)
+    {
+        var apiKey = await _apiKeyRepository.GetByIdAsync(Id);
+        await _authorizationResolver.MustHaveAccess(apiKey);
+
+        return apiKey;
+    }
+
+    public async Task<ApiKey> GetByRawAPIKey(string rawKey)
+    {
+        var hash = _apiKeyGenerator.Hash(rawKey);
+
+        return await _apiKeyRepository.GetByHashAsync(hash);
+    }
+
+    public async Task<(ApiKey key, string revealedApiKey)> CreateAsync(string name, Guid accountId)
```

Comment:
we probably don't need `accountId` param.

reason: we will never create api keys for other accounts. so probably it makes more sense to fetch logged in account here, in the service, instead of controllers passing it down here.